libretroshare/src/retroshare/rsjsonapi.h

```diff
@@ -0,0 +1,118 @@
+/*******************************************************************************
+ * libretroshare/src/retroshare: rsjsonapi.h                                   *
+ *                                                                             *
+ * Copyright (C) 2018-2019  Gioacchino Mazzurco <gio.eigenlab.org>             *
+ * Copyright (C) 2019-2019  Cyril Soler <csoler@users.sourceforge.net>         *
+ *                                                                             *
+ * This program is free software: you can redistribute it and/or modify        *
+ * it under the terms of the GNU Lesser General Public License as              *
+ * published by the Free Software Foundation, either version 3 of the          *
+ * License, or (at your option) any later version.                             *
+ *                                                                             *
+ * This program is distributed in the hope that it will be useful,             *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                *
+ * GNU Lesser General Public License for more details.                         *
+ *                                                                             *
+ * You should have received a copy of the GNU Lesser General Public License    *
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.       *
+          *
+ *******************************************************************************/
+#pragma once
+
+#include <string>
+
+class p3ConfigMgr;
+class JsonApiResourceProvider;
+
+class RsJsonAPI
+{
+public:
+    enum {
+        JSONAPI_STATUS_UNKNOWN     = 0x00,
+        JSONAPI_STATUS_RUNNING     = 0x01,
+        JSONAPI_STATUS_NOT_RUNNING = 0x02
+    };
+
+	static const uint16_t    DEFAULT_PORT = 9092 ;
+	static const std::string DEFAULT_BINDING_ADDRESS ;	// 127.0.0.1
+
+	virtual bool restart() =0;
+	virtual bool stop()  =0;
+
+	virtual void setBindingAddress(const std::string& address) =0;
+	virtual void setListeningPort(uint16_t port) =0;
+	virtual uint16_t listeningPort() const =0;
+
+    virtual void connectToConfigManager(p3ConfigMgr *cfgmgr)=0;
+
+    virtual void registerResourceProvider(const JsonApiResourceProvider *)=0;
+    virtual void unregisterResourceProvider(const JsonApiResourceProvider *)=0;
+    virtual bool hasResourceProvider(const JsonApiResourceProvider *)=0;
+
+	/**
+	 * @brief Get status of the json api server
+	 * @jsonapi{development}
+	 * @return the status picked in the enum JSONAPI_STATUS_UNKNOWN/RUNNING/NOT_RUNNING
+	 */
+	virtual int status() =0;
```

Comment:
Should be deleted like the enum upster, bool isRunning() or similar is more then enough